src/mds/Server.cc

```diff
@@ -262,6 +263,15 @@ void Server::dispatch(const cref_t<Message> &m)
     handle_client_session(ref_cast<MClientSession>(m));
     return;
   case CEPH_MSG_CLIENT_REQUEST:
+    
+    if(!mdr->session && mdr->client_request->is_queued_for_replay() && !mds->inotable->is_marked_free(inodeno_t(req->head.ino)) && 
+       ((mdr->client_request->get_op == CEPH_MDS_OP_CREATE && !mdr->has_completed) ||
+        mdr->client_request->get_op == CEPH_MDS_OP_MKNOD ||
+        mdr->client_request->get_op == CEPH_MDS_OP_MKDIR ||
+        mdr->client_request->get_op == CEPH_MDS_OP_SYMLINK)) {
+      respond_to_request(mdr, -EEXIST);
+      return;
+    }
```

Comment:
I think it's better to move these code to a place near mdcache->add_replay_ino_alloc